R/dvhydrograph-render.R

```diff
@@ -87,50 +83,58 @@ createDVHydrographPlot <- function(reportObject){
+  plot_object <- plotTimeSeries(plot_object, stat3TimeSeriesEst, 'stat3TimeSeriesEst', yLabel=yLabel, timezone, excludeZeroNegativeFlag, getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotTimeSeries(plot_object, comparisonTimeSeries, 'comparisonTimeSeries', yLabel=yLabel, timezone, excludeZeroNegativeFlag, getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotTimeSeries(plot_object, comparisonTimeSeriesEst, 'comparisonTimeSeriesEst', yLabel=yLabel, timezone, excludeZeroNegativeFlag, getDVHydrographPlotConfig, isDV=TRUE)
+
+  #Plot Other Items
+  plot_object <- plotItem(plot_object, estimated1Edges, 'estimated1Edges', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, estimated2Edges, 'estimated2Edges', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, estimated3Edges, 'estimated3Edges', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, comparisonEdges, 'comparisonEdges', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, groundWaterLevels, 'groundWaterLevels', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, fieldVisitMeasurements, 'fieldVisitMeasurements', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, minMaxPoints[['min_iv']], 'min_iv', getDVHydrographPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, minMaxPoints[['max_iv']], 'max_iv', getDVHydrographPlotConfig, isDV=TRUE)
 
   # approval bar styles are applied last, because it makes it easier to align
   # them with the top of the x-axis line
-  plot_object <- ApplyApprovalBarStyles(plot_object, dvData)
+  plot_object <- ApplyApprovalBarStyles(plot_object, approvals)
   
+  #Remove any duplicate legend items
   plot_object <- rmDuplicateLegendItems(plot_object)
   
   # custom gridlines below approval bar
   plot_object <- plot_object %>% 
     abline(v=seq(from=toStartOfDay(startDate), to=toStartOfDay(endDate), by="days"), lty=3, col="gray", where='first') %>%
     abline(v=seq(from=toStartOfDay(startDate), to=toStartOfDay(endDate), by="weeks"), col="darkgray", lwd=1, where='first')
   
-  # patch up top extent of y-axis
+  # Add space to the top of the Y Axis
   plot_object <- RescaleYTop(plot_object)
 
   #Legend
   legend_items <- plot_object$legend$legend.auto$legend
   ncol <- ifelse(length(legend_items) > 3, 2, 1)
-  leg_lines <- ifelse(ncol==2, ceiling((length(legend_items) - 6)/2), 0) 
-  legend_offset <- ifelse(ncol==2, 0.3+(0.05*leg_lines), 0.3)
-  plot_object <- legend(plot_object, location="below", cex=0.8, legend_offset=0.2, y.intersp=1.5, ncol=ncol)
+  leg_lines <- ifelse(ncol==2, ceiling((length(legend_items) - 6)/2), length(legend_items))
+  legend_offset <- ifelse(ncol==2, 0.09+(0.025*leg_lines), (0.025*leg_lines))
+  legend_offset <- ifelse(as.period(interval(startDate, endDate, tzone = attr(startDate, timezone))) < years(1), legend_offset+0.03, legend_offset+0.08)
```

Comment:
add comment about why - since there will be year labels in addition to months or something